src/types.rs

```diff
@@ -1283,6 +1290,10 @@ impl<'a, T: Asn1Readable<'a>> SequenceOf<'a, T> {
             Ok(i)
         })?;
 
+        if length < MINIMUM_LEN || length > MAXIMUM_LEN {
+            return Err(ParseError::new(ParseErrorKind::InvalidValue));
```

Comment:
This could probably use a more specific error kind

Reply:
Makes sense; something like `CollectionTooSmall`/`CollectionTooBig` (or monomorphically `InvalidCollectionSize(min, max)`).